### PR TITLE
Bratseth/more tensor doc

### PR DESCRIPTION
**Pull request by bratseth**
@lesters please review

@havardpe fyi

### CODE REVIEW

File: documentation/reference/tensor.html [hunk 1]
```diff
@@ -294,8 +360,12 @@ <h4 id="tensor">- tensor</h4>
 Examples:
 {% raw %}
 <pre>
-tensor&lt;float&gt;(x[3])(x) = {{x:0}: 0.0, {x:1}: 1.0, {x:2}: 2.0}
+t1 = tensor(x[4]):[1.0, 2.0, 3.0, 4.0]
+
+tensor&lt;float&gt;(x[3])(x) = [0.0, 1.0, 2.0]
 tensor&lt;float&gt;(x[2],y[2])(x == y) = {{x:0,y:0}: 1.0, {x:0,y:1}: 0.0, {x:1,y:0}: 0.0, {x:1,y:1}: 1.0}
+tensor(f[2])(t1{x: f + f()})) - Create a size 2 1d tensor by accessing t1 at two indexes given by adding
+                                the index of it (f - which is 0 and 1) by a function also named f.
```
Comment (lesters) [inline documentation/reference/tensor.html]: This example is perhaps a bit dense. I think the concept of generating a tensor based on another is sufficiently complex that it warrants an expanded description, with showing a few different examples with varying complexity.

Comment (bratseth) [reply documentation/reference/tensor.html]: Agreed, just wanted to capture the f and f() thing *somewhere* for now.

File: documentation/reference/tensor.html [hunk 2]
```diff
@@ -118,23 +119,71 @@ <h3 id="literal-form-examples">Literal form examples</h3>
...
+dimension in the map short form and the values of each dense subspace on the indexed short form.
+This form requires a type to be specified.</p>
+
+<b>Mixed short form example:</b>
+
+<p>
+A map of matrixes:
+<pre>
+tensor&lt;float&gt;(key{},x[2],y[3]):{ key1:[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]
+                                 key2:[[1.1, 2.1, 3.1], [4.1, 5.1, 6.1]] }
```
Comment (havardpe) [inline documentation/reference/tensor.html]: in the back-end, we require comma before key2

File: documentation/reference/tensor.html [hunk 3]
```diff
@@ -118,23 +119,71 @@ <h3 id="literal-form-examples">Literal form examples</h3>
...
+
+<p>Tensors with a single mapped dimension can be written by specifying just the label in that implicit dimension
+instead of a full address map. This form requires a type to be specified.</p>
+
+<b>Map short form example:</b>
+
+<p>
+<pre>
+tensor&lt;float&gt;(key{}):{ key1:1.0
+                       key2:2.0 }
```
Comment (havardpe) [inline documentation/reference/tensor.html]: in the back-end, we require comma before key2

File: documentation/reference/tensor.html [hunk 4]
```diff
@@ -294,8 +360,12 @@ <h4 id="tensor">- tensor</h4>
 Examples:
 {% raw %}
 <pre>
-tensor&lt;float&gt;(x[3])(x) = {{x:0}: 0.0, {x:1}: 1.0, {x:2}: 2.0}
+t1 = tensor(x[4]):[1.0, 2.0, 3.0, 4.0]
+
+tensor&lt;float&gt;(x[3])(x) = [0.0, 1.0, 2.0]
 tensor&lt;float&gt;(x[2],y[2])(x == y) = {{x:0,y:0}: 1.0, {x:0,y:1}: 0.0, {x:1,y:0}: 0.0, {x:1,y:1}: 1.0}
+tensor(f[2])(t1{x: f + f()})) - Create a size 2 1d tensor by accessing t1 at two indexes given by adding
```
Comment (havardpe) [inline documentation/reference/tensor.html]: in the back-end, we now require label expressions to have () around it

File: documentation/reference/tensor.html (hunk 2 above)
Comment (bratseth) [reply documentation/reference/tensor.html]: Yes, a typo. Fixing in a new PR - thanks.

File: documentation/reference/tensor.html (hunk 3 above)
Comment (bratseth) [reply documentation/reference/tensor.html]: Same

File: documentation/reference/tensor.html (hunk 4 above)
Comment (bratseth) [reply documentation/reference/tensor.html]: Yes, I'll add those when missing, thanks for the reminder.

**Pull request merged**